src/transformers/models/vit_mae/modeling_vit_mae.py

```diff
@@ -892,17 +982,21 @@ def unpatchify(self, patchified_pixel_values):
                 Pixel values.
         """
         patch_size, num_channels = self.config.patch_size, self.config.num_channels
-        num_patches_one_direction = int(patchified_pixel_values.shape[1] ** 0.5)
+        if not (original_img[2] == original_img[3] == 244):
+            num_patches_one_direction = original_img[2] // patch_size
+            num_patches_two_direction = original_img[3] // patch_size
+        else:
+            num_patches_one_direction = num_patches_two_direction = int(patchified_pixel_values.shape[1] ** 0.5)
         # sanity check
-        if num_patches_one_direction**2 != patchified_pixel_values.shape[1]:
+        if num_patches_one_direction * num_patches_two_direction != patchified_pixel_values.shape[1]:
             raise ValueError("Make sure that the number of patches can be squared")
 
         # unpatchify
         batch_size = patchified_pixel_values.shape[0]
         patchified_pixel_values = patchified_pixel_values.reshape(
             batch_size,
             num_patches_one_direction,
```

Comment:
let's rename here - it's a bit clearer than "one direction"

```suggestion
            num_patches_h,
```